ttnn/cpp/ttnn/operations/pool/maxpool/max_pool.hpp

```diff
@@ -173,3 +179,53 @@ Tensor maxpool2d_new(const Tensor &input,
 
 }  // namespace tt_metal
 }  // namespace tt
+
+
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+
+namespace ttnn::operations {
+namespace maxpool {
+
+using array2_t = std::array<uint32_t, 2>;
+
+// maxpool macro-op
+inline Tensor maxpool2d(const Tensor& input_tensor, uint32_t batch_size, uint32_t input_h, uint32_t input_w, uint32_t channels, array2_t kernel_size, array2_t stride, array2_t padding, array2_t dilation, Device& device) {
```

Comment:
Need to call `register_operation` to make sure that this is properly registered. 
And ideally this should be a class. You can follow the argmax example.

Reply:
I'd expect that we basically have a single op max_pool2d with multiple overloads, accepting different args. 
And single function in python, with corresponding overloads.

Reply:
There is currently a MaxPool struct which does this. This function is currently kept for backwards compatibility to support a particular version of Resnet. This Resnet version would have to ported to the new max pool, after which it can be cleaned up.

Reply:
@sankarmanoj-tt are you pushing the migration of resnet so we can cleanup the code?

Reply:
These changes to Resnet and max pool will be completely in the future as a separate PR.